app/src/main/java/com/example/androidacademy/MainActivity.kt

```diff
@@ -54,4 +62,22 @@ class MainActivity : AppCompatActivity(), ChangeFragment {
     override fun backFragmentMoviesList() {
         supportFragmentManager.popBackStack()
     }
+
+    private fun handleIntent(intent: Intent) {
+        when (intent.action) {
+            Intent.ACTION_VIEW -> {
+                val id = intent.data?.lastPathSegment?.toLongOrNull()
+                if (id != null) {
```

Comment:
Можно избавиться от переменной
```suggestion
                intent.data?.lastPathSegment?.toLongOrNull()?.let { id ->
```